numerics/hermite3_test.cpp

```diff
@@ -82,5 +95,50 @@ TEST_F(Hermite3Test, Conditioning) {
             h.EvaluateDerivative(t0_ + 19418861.806896236 * Second));
 }
 
+TEST_F(Hermite3Test, OneDimensionalInterpolationError) {
+  std::vector<std::pair<double, double>> samples;
+  for (double i = 2; i < 10; i += 1) {
+    samples.push_back({1 / i, Pow<4>(1 / i)});
+  }
+  const auto not_a_quartic =
+      Hermite3<double, double>(/*arguments=*/{0, 1},
+                               /*values=*/{0, 1},
+                               /*derivatives=*/{0, 4});
+  // |not_a_quartic| has a root at 1/2, where the error is maximal.
+  EXPECT_THAT(not_a_quartic.LInfinityError(
+      samples,
+      /*get_argument=*/[](auto&& pair) -> auto&& { return pair.first; },
+      /*get_value=*/[](auto&& pair) -> auto&& { return pair.second; }),
+      Eq(1 / 16.0));
+}
+
+TEST_F(Hermite3Test, ThreeDimensionalInterpolationError) {
+  std::vector<std::pair<Instant, Position<World>>> samples;
+  Instant const tmax = t0_ + π / 2 * Second;
+  AngularFrequency const ω = 1 * Radian / Second;
+  for (Instant t = t0_; t <= tmax; t += 1 / 32.0 * Second) {
+    samples.push_back(
+    {t, World::origin + Displacement<World>({Cos(ω * (t - t0_)) * Metre,
+                                             Sin(ω * (t - t0_)) * Metre,
+                                             0 * Metre})});
+  }
+  const auto not_a_quartic = Hermite3<Instant, Position<World>>(
+      /*arguments=*/{t0_, tmax},
+      /*values=*/
+      {World::origin + Displacement<World>({1 * Metre, 0 * Metre, 0 * Metre}),
+       World::origin + Displacement<World>({0 * Metre, 1 * Metre, 0 * Metre})},
+      /*derivatives=*/
+      {Velocity<World>(
+           {0 * Metre / Second, 1 * Metre / Second, 0 * Metre / Second}),
+       Velocity<World>(
+           {-1 * Metre / Second, 0 * Metre / Second, 0 * Metre / Second})});
+  EXPECT_THAT(
+      not_a_quartic.LInfinityError(
+          samples,
+        /*get_argument=*/[](auto&& pair) -> auto&& { return pair.first; },
+        /*get_value=*/[](auto&& pair) -> auto&& { return pair.second; }),
+      AllOf(Gt(1 * Centi(Metre)), Lt(2 * Centi(Metre))));
+}
+
```

Comment:
Should we have a test with a quartic?

Reply:
We do (it's the one above), but perhaps I shouldn't call the function that tries to match a circle `not_a_quartic`... :-p